localstack/services/awslambda/lambda_api.py

```diff
@@ -239,24 +239,39 @@ def use_docker():
     return DO_USE_DOCKER
 
 
-def process_apigateway_invocation(func_arn, path, payload, headers={},
-        resource_path=None, method=None, path_params={},
-        query_string_params={}, request_context={}):
+def get_stage_variables(api_id, stage):
+    api_gateway_client = aws_stack.connect_to_service('apigateway')
+    response = api_gateway_client.get_stage(restApiId=api_id, stageName=stage)
+    return response.get('variables', None)
+
+
+def fix_proxy_path_params(path_params):
+    proxy_path_param_value = path_params.get('proxy+')
+    if not proxy_path_param_value:
+        return
+    del path_params['proxy+']
+    path_params['proxy'] = proxy_path_param_value
+
+
+def process_apigateway_invocation(func_arn, path, payload, stage, api_id, headers={},
+                                  resource_path=None, method=None, path_params={},
+                                  query_string_params=None, request_context={}):
     try:
         resource_path = resource_path or path
+        fix_proxy_path_params(dict(path_params))
```

Comment:
Looks like we're passing a copy of the dict to the function, hence the changes are not applied to `path_params`. I think this should be instead:
```
path_params = dict(path_params)
fix_proxy_path_params(path_params)
```